bigquery_etl/metadata/validate_metadata.py

```diff
@@ -192,6 +192,33 @@ def validate_exclusion_list_expiration_days(metadata, path):
 
     return is_valid
 
+def validate_retention_policy_based_on_table_type(metadata, path):
+    """Check if any of the retention exclusion tables have expiration_days set."""
+    is_valid = True
+    table_type = metadata.labels.get("table_type") if isinstance(metadata.labels, dict) else None
+    print(f"Table type: {table_type}")
+    print(metadata)
+    expiration_days = metadata.bigquery.time_partitioning.expiration_days
+    print(f"Expiration days: {expiration_days}")
+    if expiration_days is not None and table_type == "aggregate":
+        click.echo(
+            click.style(
+                f"ERROR: Table at {path} is an aggregate table but has expiration_days set to {expiration_days}.",
+                fg="red",
+            )
+        )
+        is_valid = False
+
+    if expiration_days is None and table_type == "client_level":
```

Comment:
Should this also account for tables in the `retention_exclusion_list`, i.e. not error for them?

Reply:
Good catch! thank you @BenWu